openmc/deplete/microxs.py

```diff
@@ -0,0 +1,233 @@
+                   energy_bounds=(0, 20e6)):
+        """Generate a one-group cross-section dataframe using
+        OpenMC. Note that the ``openmc`` executable must be compiled.
+
+        Parameters
+        ----------
+        model : openmc.Model
+            OpenMC model object. Must contain geometry, materials, and settings.
+        reaction_domain : openmc.Material or openmc.Cell or openmc.Universe or openmc.RegularMesh
+            Domain in which to tally reaction rates.
+        chain_file : str
+            Path to the depletion chain XML file that will be used in depletion
+            simulation. Used to determine cross sections for materials not
+            present in the inital composition.
+        dilute_initial : float
+            Initial atom density [atoms/cm^3] to add for nuclides that
+            are zero in initial condition to ensure they exist in the cross
+            section data. Only done for nuclides with reaction rates.
+        reactions : list of str, optional
+            Reaction names to tally
+        energy_bound : 2-tuple of float, optional
+            Bounds for the energy group.
+
+        Returns
+        -------
+        MicroXS
+            Cross section data in [b]
+
+        """
+        groups = EnergyGroups(energy_bounds)
+
+        # Set up the reaction tallies
+        original_tallies = model.tallies
+        original_materials = deepcopy(model.materials)
+        tallies = Tallies()
+        xs = {}
+        reactions, diluted_materials = cls._add_dilute_nuclides(chain_file,
+                                                                model,
+                                                                dilute_initial)
+        model.materials = diluted_materials
+
+        for rxn in reactions:
+            if rxn == 'fission':
+                xs[rxn] = FissionXS(domain=reaction_domain, groups=groups, by_nuclide=True)
+            else:
+                xs[rxn] = ArbitraryXS(rxn, domain=reaction_domain, groups=groups, by_nuclide=True)
+            tallies += xs[rxn].tallies.values()
+
+        model.tallies = tallies
+
+        # create temporary run
+        with tempfile.TemporaryDirectory() as temp_dir:
+            statepoint_path = model.run(cwd=temp_dir)
+
+            with StatePoint(statepoint_path) as sp:
+                for rxn in xs:
+                    xs[rxn].load_from_statepoint(sp)
+
+        # Build the DataFrame
+        micro_xs = cls()
+        for rxn in xs:
+            df = xs[rxn].get_pandas_dataframe(xs_type='micro')
+            df.index = df['nuclide']
+            df.drop(['nuclide', xs[rxn].domain_type, 'group in', 'std. dev.'], axis=1, inplace=True)
+            df.rename({'mean': rxn}, axis=1, inplace=True)
+            micro_xs = concat([micro_xs, df], axis=1)
+
+        # Revert to the original tallies and materials
+        model.tallies = original_tallies
+        model.materials = original_materials
+
+        return cls(micro_xs)
```

Comment:
Here's a nice simplification to make -- this pulls out the 'mean' column as a `Series`, puts it in a dictionary, and then constructs the dataframe from the dict of series. Also, our standard abbreviation in OpenMC for "reaction" is `rx`
```suggestion
        series = {}
        for rx in xs:
            df = xs[rx].get_pandas_dataframe(xs_type='micro')
            series[rx] = df.set_index('nuclide')['mean']

        # Revert to the original tallies and materials
        model.tallies = original_tallies
        model.materials = original_materials

        return cls(series)
```